File: src/programmers/lv1/덧칠하기.java
package programmers.lv1;

// 최소 횟수 구하기
public class 덧칠하기 {


    private static int solution(int n, int m, int[] section) {
        int answer = 0;

        int size = section.length;
        int next = section[0]; // 다음에 칠할 구간
        int idx = 0; // 현재 위치

        while (idx < size) {
            if(next <= section[idx]) {
                next = section[idx] + m;
                answer++;
            }
            idx ++;
        }

        return answer;
    }



    public static void main(String[] args) {
        int n = 4;
        int m = 1;
        int[] section = {1, 2, 3, 4};

        int solution = solution(n, m, section);

        System.out.println(solution);
    }
}

File: kotlin_module/src/main/kotlin/programmers/lv1/덧칠하기.kt
package com.lucas.programmers.lv1

/**
 * 덧칠하기.kt: 프로그래머스 LV1
 *
 * @author: lucaskang(swings134man)
 * @since: 2025. 7. 18. 오전 1:08
 * @description:
 */
fun main() {
    println(solution_덧칠하기(8, 4, intArrayOf(2, 3, 6)))
}

fun solution_덧칠하기(n: Int, m: Int, section: IntArray): Int {
    var answer: Int = 0

    var current = 0
    for (i in section) {
        if (current < i) {
            current = i + m - 1
            answer++
        }
    }

    return answer
}